Feature1 branch

AndriyPasichnyk opened the pull request:
New part of functionallity:
 User profile with images
 Administration and User mode
 Layout changes for both modes

Commit:
feat: layout changes

Commit:
feat: allow user add User Image and see it on logged user box

Added image on Personal info screen. Added posibility select new image for user. Show image in navigation bar in logged user box.

Added button for change Administaration and User mode. Added different menus for User and Administration mode.

Added helpers for showing images and user information on navigation menu.

AndriyPasichnyk committed:
Merge branch 'feature_branch' into feature1_branch

ShoSashko commented on SportsNews/Migrations/20220531094946_AddModels02.cs:
```diff
@@ -0,0 +1,30 @@
+                {
+                    Id = table.Column<int>(nullable: false)
+                        .Annotation("SqlServer:Identity", "1, 1"),
+                    UserId = table.Column<string>(maxLength: 450, nullable: false),
```
Change UserId type from string to uniqueidentifier https://docs.microsoft.com/en-us/sql/t-sql/data-types/uniqueidentifier-transact-sql?view=sql-server-ver16

ShoSashko commented on SportsNews/Migrations/20220531094946_AddModels02.cs:
```diff
@@ -0,0 +1,30 @@
+                    Id = table.Column<int>(nullable: false)
+                        .Annotation("SqlServer:Identity", "1, 1"),
+                    UserId = table.Column<string>(maxLength: 450, nullable: false),
+                    ProfilePicture = table.Column<string>(nullable: true)
```
For Picture please try use https://quebit.com/askquebit/quebit-products/sql-server-image-and-varbinary-data-types/#:~:text=The%20IMAGE%20data%20type%20in,version%20of%20MS%20SQL%20Server.
varbinary instead of string
https://docs.microsoft.com/en-us/sql/t-sql/data-types/binary-and-varbinary-transact-sql?view=sql-server-ver16

ShoSashko commented on SportsNews/Helpers/UserInfoHelper.cs:
```diff
@@ -0,0 +1,29 @@
+            if (user.Identity.IsAuthenticated)
+            {
+                var userId = applicationDbContext.Users.FirstOrDefault(u => u.UserName == user.Identity.Name)?.Id ?? String.Empty;
+                var picture = applicationDbContext.UserPhotos.FirstOrDefault(u => u.UserId == userId)?.ProfilePicture ?? String.Empty;
```
Let's implement UoW + Repository pattern for data access. 
Remove UserInfoHelper 
Implement UnitOfWork and Repository
https://docs.microsoft.com/en-us/aspnet/mvc/overview/older-versions/getting-started-with-ef-5-using-mvc-4/implementing-the-repository-and-unit-of-work-patterns-in-an-asp-net-mvc-application

ShoSashko commented on SportsNews/Migrations/20220531094946_AddModels02.cs:
```diff
@@ -0,0 +1,30 @@
+
+namespace SportsNews.Migrations
+{
+    public partial class AddModels02 : Migration
```
Change Migration Name https://stackoverflow.com/questions/32480267/is-there-a-best-or-recommended-practice-for-naming-entity-framework-migrations

ShoSashko commented on SportsNews/Views/Shared/_Layout.cshtml:
```diff
@@ -10,39 +10,54 @@
-                    <partial name="_LoginStatusPartial" />
-                </div>
+                <a class="navbar-brand" style="width: 260px; max-height: 100px;" asp-area="" asp-controller="Home" asp-action="Index">Sports Hub</a>
+                @*<button class="navbar-toggler" type="button" data-toggle="collapse" data-target=".navbar-collapse" aria-controls="navbarSupportedContent"
```
Remove comment

ShoSashko commented on SportsNews/Helpers/UserHelper.cs:
```diff
@@ -0,0 +1,21 @@
+{
+    public static class UserHelper
+    {
+        public static HtmlString CreateUserFullName(this IHtmlHelper html, ClaimsPrincipal user)
```
This method is not creating User Full Name. 
It's getting User Full Name from Claims. 
Change naming to GetUserFullNameFromClaims

ShoSashko commented on SportsNews/Controllers/AccountController.cs:
```diff
@@ -153,5 +181,40 @@ private async Task<IdentityResult> CreateOrReplaceClaim(IdentityUser user, strin
+        private void SaveUserImage(string picture)
+        {
+            var userId = this.userManager.GetUserId(User);
+            var userPhoto = this.applicationDbContext.UserPhotos.FirstOrDefault(u => u.UserId == userId);
```
This logic will be extracted within Repository Pattern. 
Extract this method to  UserRepository.cs  class and Create Method: 
SaveUserImage

Commit:
fix: PRreview fixes

Commit:
fix: merge

ShoSashko commented on SportsNews.Data.Models/UserPhoto.cs:
```diff
@@ -0,0 +1,16 @@
+        public int Id { get; set; }
+
+        [Required]
+        [StringLength(450)]
```
We don't need that, Guid has fixed length

ShoSashko commented on SportsNews/Data/UnitsOfWork/UserPhotoUnitOfWork.cs:
```diff
@@ -0,0 +1,42 @@
+
+namespace SportsNews.Data
+{
+    public class UserPhotoUnitOfWork : IUnitOfWork
```
The unit of work class serves one purpose: to make sure that when you use multiple repositories, they share a single database context. Each repository property returns a repository instance that has been instantiated using the same database context instance as the other repository instances.

Please rename UserPhotoUnitOfWork  to UnitOfWork

ShoSashko replied:
Reread the UoW pattern

Commit:
feat: added  Info Architecture for Administarator mode

Added models for Category, Sub Category and Teams. Added repositories for these models, and added them to UoW.

Added Admin Controler with CRUD logic. Added Info Architecture page with modal elements for CRUD logic.

Added AdminMenu model for creation left menu in Administarator mode.

Commit:
fix: add changes after review

Create UoW for entire application which will contain all repositories

Fix UsePhoto model - delete noneeded attribute

Commit:
feat(InfoArchitecture): corrected Edit and added Delete to InfoArchitecture

fix in controlers refered to Commit b0e08359
